Update external-secrets docs (#4597)
kyma-project/test-infra: prow/cluster/resources/external-secrets/README.md

      version: latest # version of the GCP Secret
      property: anotherKey # name of the field in the GCP Secret JSON, unused for plain values
```

> ### Note

Reviewer: ```suggestion
>**NOTE:** The trusted and untrusted files are only applied to trusted or untrusted clusters respectively. While the workload file is applied to both trusted and untrusted clusters.
```

> The trusted/untrusted files are only applied to the trusted and untrusted clusters respectively, while the workload file is applied to both clusters.

Reviewer: ```suggestion
   The presubmit and pj-tester jobs are executed on untrusted clusters, while the periodic jobs are run on trusted clusters. Adding a Secret to the proper file allows the user to specify which type of clusters should have access to the Secret.
```

> The pre-submit and pj-tester jobs are executed on untrusted clusters, while the periodic jobs are run on the trusted clusters. Adding the secret to the proper file allows the user to specify when type of clusters should have access to the secret.

Reviewer: ```suggestion

```
